forem/forem
# 🚀 Feature: Chat channel membership manager component

## What type of PR is this? (check all applicable)


## Description
In this PR we have completed the Chat Channel Membership Manager. Which gives you more control to perform following actions as a mod on the channel.
- Ability to send invitations using an URL.
- Make other member Mod of the channel.
- Search member from a long list. 
We also fixed few bugs like 
## Related Tickets & Documents
- [#8749](https://github.com/thepracticaldev/dev.to/issues/8749) 
## QA Instructions, Screenshots, Recordings
![Screen Recording 2020-06-27 at 05 40 PM](https://user-images.githubusercontent.com/3650216/85922096-6eb63580-b89e-11ea-9ec4-00b87aeb60d6.gif)
![Screen Recording 2020-06-27 at 05 38 PM](https://user-images.githubusercontent.com/3650216/85922094-6b22ae80-b89e-11ea-865b-be1555f14672.gif)


To test:
- Set up your user to become a mod of the channel

```ruby
u = User.last # Me

cc = ChatChannel.create!(channel_type: "open", channel_name: "ACME", slug: "acme")
cc.add_users(u) # Add myself which I do see in the site, but only as a member
cc = ChatChannel.create!(channel_type: "open", channel_name: "ACME", slug: "acme")

u.chat_channel_memberships.last.update(role: "mod")
```

## Added tests?


## Added to documentation?


## [optional] Are there any post deployment tasks we need to perform?

## [optional] What gif best describes this PR or how it makes you feel?

![alt_text](https://media.giphy.com/media/MTclfCr4tVgis/giphy.gif)
